utils/time_slots_utils.js

```diff
@@ -0,0 +1,84 @@
+const {models} = require('../db/relations')
+const TimeSlot = models.TimeSlot
+
+const TimeSlotsUtils = {
+
+    getIntersection: function (slots) {
+        let res = []
+        let pointers = new Array(slots.length).fill(0)
+        let l = -1
+        let r = -1
+        while (!isNaN(l) && !isNaN(r) ) {
+            let it_slots = new Array(slots.length).fill(0)
+            for (let i = 0; i < slots.length; i++) {
+                it_slots[i] = slots[i][pointers[i]]
+            }
+            let vals = it_slots.map(slot => slot !== undefined ? slot.startDatetime.getTime() : undefined)
+            l = Math.max(...vals)
+
+            vals = it_slots.map(slot => slot !== undefined ? slot.startDatetime.getTime() + slot.duration * 60000 : undefined)
```

Comment:
Czemu * 60000?

Reply:
żeby mieć milisekundy

Reply:
Ale przecież milisekundy to najmniejszy czas jaki obsługuje node jak pamiętam więc po co to mnożyć i dlaczego przez 60000 przynajmniej bym prosił o wytłumaczenie magic number

Reply:
jak operuje na datach to wygodniej mi operować na getTime(), która zwraca milisekundy, żeby mieć końcówkę slotu potrzebuje do daty początku dodać duration => zeby się zgadzało w milisekundach, muszę przeliczyć duration, które jest w minutach na milisekundy => minuta * 60s * 1000 daje milisekundy

Reply:
To zamieniłbym to na stałą lub chociaż zapisałbym to liczbę jako
```javascript
60_000
```